libp2p/switch.nim

```diff
@@ -234,9 +235,8 @@ proc stop*(s: Switch) {.async.} =
 
   trace "Switch stopped"
 
-proc start*(s: Switch) {.async, gcsafe.} =
+proc start*(s: Switch) {.async, gcsafe, raises: [Defect, TransportListenError].} =
```

Comment:
Async raises are not supported ATM, and adding them is more error inducing than anything else

Reply:
Good catch, thanks for picking that up. The raises pragma has been removed and a TODO put in its place.